### PR TITLE
Reword license mentions

### PR DESCRIPTION
**Pull request by agarciamontoro**
#### Summary

Remove the header from the plugin settings and reword the hard requirement on an E20 license in the README.

#### Ticket Link
--

#### Checklist

- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [ ] ~~Unit tests updated~~

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,17 +4,17 @@
...
+This repository is licensed under the [Mattermost Source Available License](LICENSE). See [frequently asked questions](https://docs.mattermost.com/overview/faq.html#mattermost-source-available-license) to learn more.
 
-Although a valid Mattermost Enterprise Edition E20 license is required if using this plugin in production, the [Mattermost Source Available License](LICENSE) allows you to compile and test this plugin in development and testing environments without a Mattermost Enterprise Edition E20 license. As such, we welcome community contributions to this plugin.
+Although a valid Mattermost Enterprise Edition E20 license is required to access all features if using this plugin in production, the [Mattermost Source Available License](LICENSE) allows you to compile and test this plugin in development and testing environments without a Mattermost Enterprise Edition E20 license. As such, we welcome community contributions to this plugin.
```
Comment (justinegeffen) [inline README.md:15]:
```suggestion
Although a valid Mattermost Enterprise license is required to access all features if using this plugin in production, the [Mattermost Source Available License](LICENSE) allows you to compile and test this plugin in development and testing environments without a Mattermost Enterprise license. As such, we welcome community contributions to this plugin.
```

Comment (justinegeffen) [reply README.md:15]: Do community members need any type of license to contribute or can they be on the starter package?

File: README.md [hunk 2]
```diff
@@ -4,17 +4,17 @@
...
+If you're running an Enterprise Edition of Mattermost and don't already have a valid license, you can obtain a trial license from **System Console > Edition and License**. If you're running the Team Edition of Mattermost, including when you run the server directly from source, you may instead configure your server to enable both testing (`ServiceSettings.EnableTesting`) and developer mode (`ServiceSettings.EnableDeveloper`). These settings are not recommended in production environments. See [Contributing](#contributing) to learn more about how to set up your development environment.
```
Comment (justinegeffen) [inline README.md:17]:
```suggestion
If you're running Mattermost Professional or Mattermost Enterprise and don't already have a valid license, you can obtain a trial license from **System Console > Edition and License**. If you're running Mattermost Starter, including when you run the server directly from source, you may instead configure your server to enable both testing (`ServiceSettings.EnableTesting`) and developer mode (`ServiceSettings.EnableDeveloper`). These settings are not recommended in production environments. See [Contributing](#contributing) to learn more about how to set up your development environment.
```

File: plugin.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
         "bundle_path": "webapp/dist/main.js"
     },
     "settings_schema": {
-        "header": "Incident Collaboration requires a Mattermost Cloud or Mattermost E20 License.",
+        "header": "",
```
Comment (justinegeffen) [inline plugin.json:21]: Incident Collaboration requires a Mattermost Professional or Mattermost Enterprise license.

**Review by justinegeffen — COMMENTED** [review]
Included new package names.

**Comment by justinegeffen** [comment]
@jasonblais: Just confirming that the terminology is correct (i.e. "Mattermost Professional" as opposed to "The Professional Edition of Mattermost".)

File: README.md (hunk 1 above)
Comment (agarciamontoro) [reply README.md:15]: They can be on the Starter package, or even in the Team Edition. If they enable the development and testing modes in the system console, they get access to all features. I like your suggestion.

File: plugin.json (hunk 3 above)
Comment (agarciamontoro) [reply plugin.json:21]: I'd say we go with an empty header here. This is only shown under the plugin settings, in the system console, so I don't think we need any description here.

File: README.md (hunk 2 above)
Comment (agarciamontoro) [reply README.md:17]: If you're running an Enterprise Edition of Mattermost and don't already have a valid license <- This means the Starter Edition, not the professional one. We can change it to that.

I think we still have the Team Edition (don't know if its name changed, though), which is functionally identical to Starter, but technically different (they're different binaries: in Team Edition, you cannot add a license; in Starter, you can upgrade to Professional or Enterprise by just updating the license).

**Review by iomodo — APPROVED** [review]

**Review by lieut-data — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (justinegeffen) [reply README.md:17]: Ah I see what you mean. Thanks for clarifying that!

**Review by justinegeffen — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (agarciamontoro) [reply README.md:17]: @justinegeffen I am re-reading this and I don't know what I meant yesterday. Your suggestion is perfectly valid. Sorry for that!

File: README.md [hunk 4]
```diff
@@ -4,17 +4,17 @@
 [![Code Coverage](https://img.shields.io/codecov/c/github/mattermost/mattermost-plugin-incident-collaboration/master.svg)](https://codecov.io/gh/mattermost/mattermost-plugin-incident-collaboration)
 [![Release](https://img.shields.io/github/v/release/mattermost/mattermost-plugin-incident-collaboration)](https://github.com/mattermost/mattermost-plugin-incident-collaboration/releases/latest)
 
- Mattermost Incident Collaboration allows your team to coordinate, manage, and resolve incidents from within Mattermost. For configuration and administration information visit our [documentation](https://docs.mattermost.com/administration/devops-command-center.html). 
+Mattermost Incident Collaboration allows your team to coordinate, manage, and resolve incidents from within Mattermost. For configuration and administration information visit our [documentation](https://docs.mattermost.com/administration/devops-command-center.html).
 
 ![Mattermost Incident Collaboration](assets/incident_response.png)
 
 ## License
```
Comment (jasonblais) [inline README.md:11]: @jamiehurewitz when you're reviewing the source available license text, are you also open to reviewing this License section of the README file to ensure we're not saying anything incorrectly?

**Comment by jasonblais** [comment]
"Mattermost Professional" is correct, but I've asked Jamie to review the text as part of her other license reviews for this repackaging effort.

**Comment by agarciamontoro** [comment]
@jasonblais, @jamiehurewitz, do you think we can make that review after we merge this PR? Or should we wait?

File: README.md (hunk 4 above)
Comment (jamiehurewitz) [reply README.md:11]: Yes.

Comment (jamiehurewitz) [reply README.md:11]: @jasonblais @agarciamontoro I think it is ok to go ahead and make the replacement terms, I can just incorporate that as part of my review.

Comment (agarciamontoro) [reply README.md:11]: @jamiehurewitz, cool, thank you! Then I'll merge this. Please ping us if you need anything from us :)

**Pull request merged**